gateway-ha/src/main/java/io/trino/gateway/ha/HaGatewayLauncher.java

```diff
@@ -94,6 +86,18 @@ private void start(List<Module> additionalModules, HaGatewayConfiguration config
         logger.info("======== SERVER STARTED ========");
     }
 
+    private void handleConfigurationException(ApplicationConfigurationException e)
```

Comment:
It's hard to imagine what happens from this method name. We could rename it to `logConfigurationException` and move `System.exit(100)` to the caller.